release/nightly_tests/dataset/parquet_data_generator.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+import pandas as pd
+import numpy as np
+
+import ray
+
+#
+# Data generation utilities for the parquet files.
+#
+
+
+def generate_data(num_rows, num_files, num_row_groups_per_file, data_dir):
+    results = []
+    for file_index, global_row_index in enumerate(
+        range(0, num_rows, num_rows // num_files)
+    ):
+        num_rows_in_file = min(num_rows // num_files, num_rows - global_row_index)
+        results.append(
+            generate_file.remote(
+                file_index,
+                global_row_index,
+                num_rows_in_file,
+                num_row_groups_per_file,
+                data_dir,
+            )
+        )
+    filenames, data_sizes = zip(*ray.get(results))
+    return filenames, sum(data_sizes)
+
+
+@ray.remote
+def generate_file(
+    file_index, global_row_index, num_rows_in_file, num_row_groups_per_file, data_dir
+):
+    buffs = []
+    for group_index, group_global_row_index in enumerate(
+        range(0, num_rows_in_file, num_rows_in_file // num_row_groups_per_file)
+    ):
+        num_rows_in_group = min(
+            num_rows_in_file // num_row_groups_per_file,
+            num_rows_in_file - group_global_row_index,
+        )
+        buffs.append(
+            generate_row_group(group_index, group_global_row_index, num_rows_in_group)
+        )
+    df = pd.concat(buffs)
+    data_size = df.memory_usage(deep=True).sum()
+    filename = os.path.join(data_dir, f"input_data_{file_index}.parquet.snappy")
+    df.to_parquet(
+        filename,
+        compression="snappy",
```

Comment:
ZSTD is the new industry standard for Parquet/ORC compression (e.g. Meta and [Apple](https://www.databricks.com/session_na21/the-rise-of-zstandard-apache-spark-parquet-orc-avro) are using it). Can we have benchmark for ZSTD as well?

Reply:
Yep, zstd is much densely compressed and hence hurts read performance, which makes it good to capture in benchmark.